kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -198,14 +198,105 @@ class KRATOS_API(KRATOS_CORE) CalculateEmbeddedNodalVariableFromSkinProcess : pu
     ///@name Life Cycle
     ///@{
 
+    /**
+     * @brief Get the Default Settings object
+     * This method returns the default parameters for this proces.
+     * Note that it is required to be static since it is called during
+     * the construction of the object so no instantation exists yet.
+     * @return Parameters Default parameters json string
+     */
+    static Parameters GetDefaultSettings()
+    {
+        Parameters default_settings(R"(
+        {
+            "base_model_part_name": "",
+            "skin_model_part_name": "",
+            "skin_variable_name": "",
+            "embedded_nodal_variable_name": "",
+            "level_set_type": "",
+            "buffer_position": 0,
+            "aux_model_part_name": "IntersectedElementsModelPart",
+            "linear_solver_settings": {
+                "preconditioner_type": "amg",
+                "solver_type": "amgcl",
+                "smoother_type": "ilu0",
+                "krylov_type": "cg",
+                "max_iteration": 1000,
+                "verbosity": 0,
+                "tolerance": 1e-8,
+                "scaling": false,
+                "block_size": 1,
+                "use_block_matrices_if_possible": true
+            }
+        }
+        )");
+
+        return default_settings;
+    }
+
+    /**
+     * @brief Construct a new Calculate Embedded Nodal Variable From Skin Process object
+     * Constructor with model and json settings
+     * @param rModel Model container
+     * @param rSettings Settings json string
+     */
+    CalculateEmbeddedNodalVariableFromSkinProcess(
+        Model &rModel,
+        Parameters rSettings)
+        : CalculateEmbeddedNodalVariableFromSkinProcess(
+            rModel.GetModelPart(rSettings["base_model_part_name"].GetString()),
+            rModel.GetModelPart(rSettings["skin_model_part_name"].GetString()),
+            [] (Parameters x) -> Parameters {x.ValidateAndAssignDefaults(GetDefaultSettings()); return x;} (rSettings))
+    {
+    }
+
+    /**
+     * @brief Construct a new Calculate Embedded Nodal Variable From Skin Process object
+     * Constructor with background and skin model parts as well as json settings.
+     * @param rBaseModelPart Background mesh model part reference
+     * @param rSkinModelPart Embedded skin model part reference
+     * @param rSettings Settings json string
+     */
     CalculateEmbeddedNodalVariableFromSkinProcess(
         ModelPart &rBaseModelPart,
         ModelPart &rSkinModelPart,
+        Parameters rSettings)
+        : CalculateEmbeddedNodalVariableFromSkinProcess(
```

Comment:
if you do this you are skipping the validation...

Reply:
I am aware of this... The point is that I don't know how to check the input settings in this constructor but avoiding to check them when it is called by the model one (basically to avoid checking the settings twice). At the end, this constructor is an auxiliary tool to avoid checking the defaults more than one. Maybe it makes sense to make it private...

Reply:
Finally I made it protected in case it is needed by eventual (Trilinos) derived classes.